sdk/src/Core/Amazon.Runtime/AmazonServiceClient.cs

```diff
@@ -55,14 +56,14 @@ protected virtual bool SupportResponseLogging
         {
             add
             {
-                lock (this)
+                lock (_lock)
```

Comment:
💡 This lock would not be necessary if the `mBeforeMarshallingEvent` field was removed and the event declaration simplified to:

```csharp
internal event PreRequestEventHandler BeforeMarshallingEvent;
```